dags/map_reproducibility/utils/common_utils_2.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"k8s python api helper for lease object"
+
+import logging
+
+from airflow.decorators import task
+from kubernetes import client as k8s_client
+from kubernetes import config
+from kubernetes.client.exceptions import ApiException
+from http import HTTPStatus
+
+# Configure logging
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+@task.sensor(poke_interval=60, mode="reschedule")
+def create_lease(lease_name: str, holder_identity: str = "airflow") -> bool:
+  """
+  Creates a Kubernetes Lease object using the Python client.
+
+  Args:
+    lease_name (str): The name of the lease to create.
+    holder_identity (str): The identity of the holder of the lease. Defaults to "airflow".
+
+  Returns:
+    bool: True if the lease was created, False if it already existed.
+  """
+  config.load_kube_config()
+  api = k8s_client.CoordinationV1Api()
+  body = k8s_client.V1Lease(
+      api_version="coordination.k8s.io/v1",
+      kind="Lease",
+      metadata=k8s_client.V1ObjectMeta(
+          name=lease_name,
+          namespace="default",
+      ),
+      spec=k8s_client.V1LeaseSpec(
+          holder_identity=holder_identity,
+      ),
+  )
+
+  try:
+    api.create_namespaced_lease("default", body)
+    logger.info(f"Successfully created lease: {lease_name}")
+    return True
+  except ApiException as e:
+    if e.status == HTTPStatus.CONFLICT:
+      logger.warning(
+          f"Lease '{lease_name}' already exists. Retrying after one minute..."
+      )
+      logger.info(f"Status: {e.status}, Reason: {e.reason}, Body: {e.body}")
+      return False
+    else:
+      logger.error(
+          f"Failed to create lease. Status: {e.status}, Reason: {e.reason}, Body: {e.body}"
+      )
+
+  return True
+
+
+@task
+def delete_lease(lease_name: str) -> bool:
```

Comment:
```suggestion
def delete_lease(name: str):
```
note: the returned value is not in used by caller (DAG)